experimental/e2e/storage/har.go

```diff
@@ -222,12 +219,13 @@ func (s *HARStorage) Entries() []*Entry {
 	return entries
 }
 
-// Delete removes the HAR entry with the given ID.
-func (s *HARStorage) Delete(id string) bool {
-	s.lock.Lock()
-	defer s.lock.Unlock()
-	for i, e := range s.har.Log.Entries {
-		if e.Comment == id {
+// Delete removes the HAR entry matching the given Request.
+func (s *HAR) Delete(req *http.Request) bool {
+	for i, e := range s.Entries() {
+		if res := e.Match(req); res != nil {
```

Comment:
Nit: Is probably just me, but the nested matching gets a little confusing to read - maybe because all the signatures look the same. I wonder if there's a nicer way to differentiate just for clarity sake

Reply:
i think this was because i was trying to find a way to get the index of the entry without adding it to `Match`. i added a separate method called `findEntry` in [0f4efb1](https://github.com/grafana/grafana-plugin-sdk-go/pull/473/commits/0f4efb175948bfb4075dda4a5d3a740611bd6921) that satisfies the use case for both `Match` and `Delete`. does that make things clearer, or did you mean the nesting of `fixture.Match` -> `har.Match` -> `entry.Match`?